Exiled.API/Features/Lockers/Chamber.cs

```diff
@@ -0,0 +1,174 @@
+        /// </summary>
+        /// <remarks>
+        /// Will be used if <see cref="UseMultipleSpawnpoints"/> is <see langword="true"/>.
+        /// </remarks>
+        public IEnumerable<Transform> Spawnpoints
+        {
+            get => Base._spawnpoints;
+            set => Base._spawnpoints = value.ToArray();
+        }
+
+        /// <summary>
+        /// Gets or sets the list with <see cref="ItemType"/> which can be spawned in this chamber.
+        /// </summary>
+        public IEnumerable<ItemType> AcceptableTypes
+        {
+            get => Base.AcceptableItems;
+            set => Base.AcceptableItems = value.ToArray();
+        }
+
+        /// <summary>
+        /// Gets or sets required permissions to open this chamber.
+        /// </summary>
+        public KeycardPermissions RequiredPermissions
+        {
+            get => (KeycardPermissions)Base.RequiredPermissions;
+            set => Base.RequiredPermissions = (Interactables.Interobjects.DoorUtils.KeycardPermissions)value;
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether or not multiple spawnpoint system should be used.
+        /// </summary>
+        /// <remarks>
+        /// If true, spawnpoint system will use <see cref="Spawnpoints"/>. Otherwise, <see cref="Spawnpoint"/>.
+        /// </remarks>
+        public bool UseMultipleSpawnpoints
+        {
+            get => Base._useMultipleSpawnpoints;
+            set => Base._useMultipleSpawnpoints = value;
+        }
+
+        /// <summary>
+        /// Gets or sets a spawnpoint for items in locker chamber.
+        /// </summary>
+        /// <remarks>
+        /// Will be used if <see cref="UseMultipleSpawnpoints"/> is <see langword="false"/>.
+        /// </remarks>
+        public Transform Spawnpoint
+        {
+            get => Base._spawnpoint;
+            set => Base._spawnpoint = value;
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether or not items should be spawned as soon as they are initialised.
+        /// </summary>
+        public bool InitiallySpawn
+        {
+            get => Base._spawnOnFirstChamberOpening;
+            set => Base._spawnOnFirstChamberOpening = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the amount of time before a player can interact with the chamber again.
+        /// </summary>
+        public float Cooldown
+        {
+            get => Base._targetCooldown;
+            set => Base._targetCooldown = value;
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether or not the chamber can currently be interacted with.
+        /// </summary>
+        public bool CanInteract => Base.CanInteract;
+
+        /// <summary>
+        /// Gets the locker where this chamber is located.
+        /// </summary>
+        public Locker Locker => Locker.Get(x => x.Chambers.Contains(this)).FirstOrDefault();
```

Comment:
i have some doubt if this is really the best way to get the Locker.

Reply:
Idk really. Couldn't find better way